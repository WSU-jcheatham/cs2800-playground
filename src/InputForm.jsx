import { React } from 'https://unpkg.com/es-react/dev';

/**
 * A form that has a labeled input and can save a text value.
 *
 * Takes 3 props:
 *   - value - the current input value
 *   - setValue - a function that takes a new input value (a string)
 *   - saveName - a function that takes no arguments that is called when the
 *   user wants to save the current input value
 */
export default function InputForm(props) {
  const { value, setValue, saveName } = props;

  return (
    <form className="input-form">
      <label>
        Name{' '}
        <input value={value} onChange={(event) => setValue(event.target.value)} />
      </label>
      <button type="button" onClick={() => saveName(value)}>Submit</button>
    </form>
  );
}
