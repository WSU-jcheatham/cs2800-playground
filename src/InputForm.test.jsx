import { React } from 'https://unpkg.com/es-react/dev';
import { render, fireEvent } from '@testing-library/react';
import InputForm from './InputForm.jsx';

test('InputForm renders', () => {
  const setValue = () => undefined;

  const { container } = render(<InputForm setValue={setValue} />);
  expect(container.querySelectorAll('input')).toHaveLength(1);
  expect(container.querySelectorAll('button')).toHaveLength(1);
  expect(container.querySelector('label')).toHaveTextContent('Name');
  expect(container).toMatchSnapshot();
});

test('InputForm saves its input', () => {
  const setValue = () => undefined;
  const saveName = jest.fn();

  const { container } = render(
    <InputForm saveName={saveName} value="foo" />
  );

  expect(saveName).not.toHaveBeenCalled();

  const button = container.querySelector('button');
  fireEvent.click(button);

  expect(saveName).toHaveBeenCalledTimes(1);
  expect(saveName).toHaveBeenCalledWith('foo');
});
