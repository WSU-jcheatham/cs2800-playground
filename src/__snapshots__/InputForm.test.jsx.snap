// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputForm renders 1`] = `
<div>
  <form
    class="input-form"
  >
    <label>
      Name
       
      <input
        value=""
      />
    </label>
    <button
      type="button"
    >
      Submit
    </button>
  </form>
</div>
`;
